### ISSUE TITLE
Different casting on creation and retrieving from db

### ISSUE THREAD

**Issue by KernelMadness** [elixirmoney/money#116]
I think this [commit](https://github.com/elixirmoney/money/commit/4e633b590fc42b2d110996dbfa1dda8c66036ab9) introduced a bug.

```
MyApp.Records.create_record(%{amount: "$42"})

{:ok,
 %MyApp.Records.Records{
   __meta__: #Ecto.Schema.Metadata<:loaded, "records">,
   amount: 4200,
   id: 1,
   inserted_at: ~N[2019-10-02 16:30:32],
   updated_at: ~N[2019-10-02 16:30:32]
 }}
```

amount is `4200`

```
MyApp.Records.list_records()

[
  %MyApp.Records.Record{
    __meta__: #Ecto.Schema.Metadata<:loaded, "records">,
    amount: %Money{amount: 4200, currency: :USD},
    id: 1,
    inserted_at: ~N[2019-10-02 16:30:32],
    updated_at: ~N[2019-10-02 16:30:32]
  }
]
```

amount is `%Money{amount: 4200, currency: :USD}`

This behaviour makes tests like this always fail

```
test "get_record!/1 returns the record with given id" do
  record = record_fixture()
  assert Records.get_record!(record.id) == record
end
```

**Comment by joshprice** [comment]
Have seen this as well in our test suite

**Comment by cdvv7788** [comment]
Maybe related to https://github.com/elixirmoney/money/issues/118

**Comment by Nitrino** [comment]
This is a hack to get validations to work, but I agree with you, the structures after casting and loading from the database, should be same. 
So this is a bug and we should abandon validations in favour of valid cast

**Comment by Nitrino** [comment]
Fixed in v1.6.1
